### PR TITLE
2nd Assignment by 202401743 (Dayeon Lee)

### PR DESCRIPTION
**Pull request by ldy56-hub**
# Assignment Submission

## Student Information
- **Student ID**: [Your Student ID]
- **Name**: [Your Full Name]
- **Assignment**: [Assignment Number]

## Changed Files

```
submissions
└── 202401743
    └── assignment2
        ├── problem1
        │   ├── README.md
        │   └── main.py
        ├── problem2
        │   ├── README.md
        │   └── textops
        │       ├── __init__.py
        │       ├── clean
        │       │   ├── __init__.py
        │       │   └── filters.py
        │       ├── main.py
        │       └── tokenize
        │           ├── __init__.py
        │           └── word.py
        ├── problem3
        │   ├── README.md
        │   └── main.py
        ├── problem4
        │   ├── README.md
        │   └── dsops
        │       ├── __init__.py
        │       ├── main.py
        │       ├── split
        │       │   ├── __init__.py
        │       │   └── train_test.py
        │       └── stats
        │           ├── __init__.py
        │           └── labels.py
        ├── problem5
        │   ├── README.md
        │   └── cachekit
        │       ├── __init__.py
        │       └── main.py
        └── problem6
            ├── README.md
            └── main.py
```

### Problem 1
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 2
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 3
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 4 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 5 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 6 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

## Testing
- [ ] I have run `python grader.py` locally
- [ ] All implemented problems pass their tests
- [ ] I have tested edge cases mentioned in the problem statements

## Additional Notes

---
**Note**: This PR will be automatically validated for:
- Correct directory structure
- Valid Python syntax
- PR title format: `Nth Assignment by {student_id} (Full Name)`
- File organization compliance

### CODE REVIEW

File: submissions/202401743/assignment2/problem2/textops/clean/filters.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import re
+import string
+
+def clean_text(s: str) -> str:
+   s = s.lower()
+   s = re.sub(r"\s+", " ", s)
+   s = s.strip()
+   keep = {"'", "-"}
+   remove = set(string.punctuation) - keep
+   punctuation_pattern = "[" + re.escape("".join(remove)) + "]"
```
Comment (hist0613) [inline submissions/202401743/assignment2/problem2/textops/clean/filters.py:10]: 오 이 라인은 좀 어려운데 주석을 달아주시겠어요?

**Comment by hist0613** [comment]
본문에 학번 등 정보 채워주세요!

File: submissions/202401743/assignment2/problem2/textops/clean/filters.py (hunk 1 above)
Comment (ldy56-hub) [reply submissions/202401743/assignment2/problem2/textops/clean/filters.py:10]: s = s.strip() # 공백제거
   keep = {"'", "-"} #유지할 문자 set으로 정의
   remove = set(string.punctuation) - keep  
   # string.punctuation : 표준 문장부호 포함 문자열  -> set으로 변환 
   # => 제거할 문자열들만을 포함하는 remove set 생성
   punctuation_pattern = "[" + re.escape("".join(remove)) + "]"
   # "".join() : remove내의 요소들을 연결 
   # re.escape() : 다른 의미를 가진 문자들을 다른 문자들과 같이 평범한 문자로 취급
   # punctuation_pattern = [제거할 문자열]

filters.py에서 변경했습니다..!
